sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala

```diff
@@ -349,6 +357,17 @@ private[hive] class SparkExecuteStatementOperation(
     }
   }
 
+  def timeoutCancel(): Unit = {
+    synchronized {
+      if (!getStatus.getState.isTerminal) {
+        logInfo(s"Timeout and Cancel query with $statementId ")
+        cleanup()
+        setState(OperationState.TIMEDOUT)
```

Comment:
nit: `logInfo` is located just before `HiveThriftServer2.eventManager.onXXX`?
```
        cleanup()
        setState(OperationState.TIMEDOUT)
        logInfo(s"Timeout and Cancel query with $statementId ")
        HiveThriftServer2.eventManager.onStatementCanceled(statementId)
```
https://github.com/apache/spark/blob/42f01e314b4874236544cc8b94bef766269385ee/sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkOperation.scala#L51-L52

Reply:
thanks, fixed

Reply:
> Please setState before cleanup. It's an open bug, see #28912

Yea, nice suggestion! Might be better to add tests for that case, too.

Reply:
> Please setState before cleanup. It's an open bug, see #28912

Fixed, thanks for your review